modules/etcpak/image_decompress_etcpak.cpp

```diff
@@ -0,0 +1,197 @@
+			if (incomplete) {                                                                                                                         \
+				for (uint32_t cy = 0; cy < yblock; cy++) {                                                                                            \
+					for (uint32_t cx = 0; cx < xblock; cx++) {                                                                                        \
+						memcpy(&dec_blocks[(y + cy) * 4 * width + (x + cx) * color_bytesize], &output[cy * 4 + cx * color_bytesize], color_bytesize); \
+					}                                                                                                                                 \
+				}                                                                                                                                     \
+			}                                                                                                                                         \
+		}                                                                                                                                             \
+	}
+
+	if (width % 4 != 0 || height % 4 != 0) {
+		uint64_t src_pos = 0;
+
+		uint8_t rgba8_output[4 * 4 * 4];
+
+		switch (format) {
+			case Etcpak_R: {
+				DECOMPRESS_LOOP_SAFE(DecodeRBlock, ETCPAK_R_BLOCK_SIZE, 4, rgba8_output)
+			} break;
+			case Etcpak_RG: {
+				DECOMPRESS_LOOP_SAFE(DecodeRGBlock, ETCPAK_RG_BLOCK_SIZE, 4, rgba8_output)
+			} break;
+			case Etcpak_RGB: {
+				DECOMPRESS_LOOP_SAFE(DecodeRGBBlock, ETCPAK_RGB_BLOCK_SIZE, 4, rgba8_output)
+			} break;
+			case Etcpak_RGBA: {
+				DECOMPRESS_LOOP_SAFE(DecodeRGBABlock, ETCPAK_RGBA_BLOCK_SIZE, 4, rgba8_output)
+			} break;
+		}
+
+	} else {
+		uint64_t src_pos = 0, dst_pos = 0;
+
+		switch (format) {
+			case Etcpak_R: {
+				DECOMPRESS_LOOP(DecodeRBlock, ETCPAK_R_BLOCK_SIZE, 4)
+			} break;
+			case Etcpak_RG: {
+				DECOMPRESS_LOOP(DecodeRGBlock, ETCPAK_RG_BLOCK_SIZE, 4)
+			} break;
+			case Etcpak_RGB: {
+				DECOMPRESS_LOOP(DecodeRGBBlock, ETCPAK_RGB_BLOCK_SIZE, 4)
+			} break;
+			case Etcpak_RGBA: {
+				DECOMPRESS_LOOP(DecodeRGBABlock, ETCPAK_RGBA_BLOCK_SIZE, 4)
+			} break;
+		}
+	}
+
+#undef DECOMPRESS_LOOP
+#undef DECOMPRESS_LOOP_SAFE
+}
+
+void _decompress_etc(Image *p_image) {
+	uint64_t start_time = OS::get_singleton()->get_ticks_msec();
+
+	int width = p_image->get_width();
+	int height = p_image->get_height();
+
+	// Compressed images' dimensions should be padded to the upper multiple of 4.
+	// If they aren't, they need to be realigned (the actual data is correctly padded though).
+	if (width % 4 != 0 || height % 4 != 0) {
+		int new_width = width + (4 - (width % 4));
+		int new_height = height + (4 - (height % 4));
+
+		print_verbose(vformat("Compressed image's dimensions are not multiples of 4 (%dx%d), aligning to (%dx%d)", width, height, new_width, new_height));
```

Comment:
Can we print the image's resource path here if available?

Like this: https://github.com/godotengine/godot/blob/1aaf20b1f1f3ab59f04db74b166540327a3f1f90/editor/import/resource_importer_texture.cpp#L125